### ISSUE TITLE
GameGuru MAX - Focus or Zoom To Selected Item Request

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [Dark-Basic-Software-Limited/GameGuruRepo#1567]
If this doesn't already exist, I'd like to see a function added to be able to zoom to or focus on a selected item. This could be via a simple key press after selecting an entity in your scene. So, for example, let's say you have a large level and you want to work on or add things to a particular area, but your camera is not close to that area, even if you can see it off in the distance. You could then click on an entity and press the key. The camera would then relocate to be near that entity, allowing the end-user to instantly get to work there.

I've also seen this function so that when the "focus" key is pressed, the view port camera orbits the selected entity. In other words, while the key is pressed, moving the mouse orbits around the entity. This makes it a lot easier to finetune things like positioning and to view an entity easily from all sides.

**Comment by rvanner** [comment]
@ArgentArts Yes, it's a needed item. Maybe a double click on the terrain or object when you are a certain distance away will bring you to that point.

**Comment by ArgentArts** [comment]
I just learned that when you have the detailed object list open, you can double-click on an item there, and it zooms to that item in the scene. So, that's a start. What I think should happen, too, is that if you select an item in the view port, it should also highlight it in the Detailed Object List. The list, as it stands, is okay ... but we can't do much with it (like rename things on the list). As a result, we can't know what #13 Light is compared to #40 Light, etc.

Being able to orbit a focused item via a key press and mouse move (as mentioned above) is pretty handy. I use it all the time in other software (both modeling and game creation).

**Comment by rvanner** [comment]
@ArgentArts Could you make a small video that shows how other software deals with this? I'd like to see how they solved the issue.

**Comment by ArgentArts** [comment]
Which? The key press + mouse movement to orbit a selected entity? Or selecting an entity in the view port and also having it selected in the entity list?

**Comment by rvanner** [comment]
Please record all you can. Extra info will help us I'm sure of it.

**Comment by ArgentArts** [comment]
Here's a video. The first thing it shows is a scene with nothing selected, the view port to the right and the scene list to the left. Once I click on the "Wall_A" object, it highlights, but so does it's name in the list. After that, I press and hold a key (ALT, in this case) while the entity is still selected. Doing so centers the item and as I move the mouse side to side, I orbit the entity so I can easily see it from all sides. Then I release the key and my mouse goes back to the standard way of moving the camera and I can use the WASD keys to move about. Pressing the key (ALT) again, brings the selected entity back to center and the mouse orbits the entity. This particular program uses ALT, another I use uses "F".

https://user-images.githubusercontent.com/36735507/142447363-c1362795-b2f3-4b60-b0de-ecf7ad729bf3.mp4

**Comment by rvanner** [comment]
@ArgentArts Thanks for sharing this. I agree that we should highlight the object in the Level Objects when it's been clicked on and the use of Alt to then spin around the object is a nice feature and something missing in MAX.

@LeeBamberTGC Take a look at this. I think we should add both to MAX.

**Comment by ArgentArts** [comment]
You're very welcome. Glad I could help. :)
